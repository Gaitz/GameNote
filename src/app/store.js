import { configureStore } from "@reduxjs/toolkit"
import authenticationReducer from "game-note/features/authentication/authenticationSlice"

const store = configureStore({
    reducer: { authentication: authenticationReducer }
})

export default store
