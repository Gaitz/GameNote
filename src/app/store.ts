import { configureStore } from "@reduxjs/toolkit"
import counterReducer from "game-note/features/example/couter/counterSlice"

const store = configureStore({
  reducer: { counter: counterReducer }
})

export default store
export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
