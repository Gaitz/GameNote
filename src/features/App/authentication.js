const SIGN_IN = "game-note/authentication/SIGN_IN"
const SIGN_OUT = "game-note/authentication/SIGN_OUT"

const initialAuthentication = {
  isSignIn: false,
  user: null,
}

export default function reducer(
  state = initialAuthentication,
  action = { type: undefined }
) {
  switch (action.type) {
    case SIGN_IN:
      return { ...state, isSignIn: true, user: action.user }
    case SIGN_OUT:
      return { ...state, isSignIn: false, user: null }
    default:
      return state
  }
}

export function userSignIn(user) {
  return { type: SIGN_IN, user: user }
}

export function userSignOut() {
  return { type: SIGN_OUT }
}
