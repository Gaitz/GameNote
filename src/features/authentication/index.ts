export * from "./Login";
export * from "./authenticationSlice"
export * from "./hooks"
export * from "./EmailLogin"
