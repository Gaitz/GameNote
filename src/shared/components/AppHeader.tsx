import { AppBar, Toolbar, Typography, IconButton } from "@material-ui/core"
import { makeStyles } from "@material-ui/core/styles";
import MenuIcon from "@material-ui/icons/Menu"

const useStyles = makeStyles(() => ({
  menuButton: {
    marginLeft: "auto"
  }
}));

export default function AppHeader () {
  const classes = useStyles();

  return (
    <AppBar position="static">
      <Toolbar>
        <Typography>
          Game Note
        </Typography>
        <IconButton className={classes.menuButton} edge="end" color="inherit" aria-label="menu">
          <MenuIcon />
        </IconButton>
      </Toolbar>
    </AppBar>
  )
}
