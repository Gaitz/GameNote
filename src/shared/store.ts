import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux"
import { configureStore } from "@reduxjs/toolkit"
import authenticationSlice from "game-note/features/authentication/authenticationSlice"

const store = configureStore({
  reducer: { authentication: authenticationSlice }
})

export default store

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch

export const useAppDispatch = () => useDispatch<AppDispatch>()
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
