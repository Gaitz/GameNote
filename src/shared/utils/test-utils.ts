import { render } from "@testing-library/react"
import store from "game-note/shared/store"
import theme from "game-note/shared/styles/theme"

// const AllTheProviders = ({ children }) => {
//   return (
//     <ReduxProvider store={store}>
//       <ChakraProvider theme={theme}>{children}</ChakraProvider>
//     </ReduxProvider>
//   )
// }

// const customRender = (ui: ReactElement, options?: object) =>
//   render(ui, { wrapper: AllTheProviders, ...options })

// // re-export everything
// export * from "@testing-library/react"

// // override render method
// export { customRender as render }
