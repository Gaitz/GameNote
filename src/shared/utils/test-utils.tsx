import "@testing-library/jest-dom"
import { render } from "@testing-library/react"
import { Provider as ReduxProvider } from "react-redux"
import { ChakraProvider } from "@chakra-ui/react"
import store from "game-note/shared/store"
import theme from "game-note/shared/styles/theme"
import React, { FC } from "react"
import type { ReactElement } from "react"
import type { RenderOptions } from "@testing-library/react"

/* eslint-disable react/prop-types */
const AllTheProviders: FC = ({ children }) => {
  return (
    <ReduxProvider store={store}>
      <ChakraProvider theme={theme}>{children}</ChakraProvider>
    </ReduxProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
) => render(ui, { wrapper: AllTheProviders, ...options })

export * from "@testing-library/react"
export { default as userEvent } from "@testing-library/user-event"

export { customRender as render }
