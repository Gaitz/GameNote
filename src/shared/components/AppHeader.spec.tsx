import { fireEvent, render } from "game-note/shared/utils/test-utils"
import store from "game-note/shared/store"
import AppHeader from "./AppHeader"
import { signIn } from "game-note/features/authentication"

describe("AppHeader component", () => {
  it("Snapshot", () => {
    const { asFragment } = render(<AppHeader />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("Sign out", async () => {
    const dispatch = store.dispatch

    const testDisplayName = "test user name"
    const testEmail = "test@test.test"
    const testPhotoUrl = "https://avatars.githubusercontent.com/u/5541565?v=4"

    const { getByTestId, findByTestId, asFragment } = render(<AppHeader />)

    dispatch(
      signIn({
        displayName: testDisplayName,
        email: testEmail,
        photoURL: testPhotoUrl
      })
    )

    const menuButton = getByTestId("appHeaderMenuButton")
    expect(menuButton).toBeEnabled()

    fireEvent.click(menuButton)
    let userEmail = getByTestId("userEmail")
    expect(userEmail).toHaveTextContent(testEmail)

    const signOutButton = getByTestId("signOutButton")
    fireEvent.click(signOutButton)
    userEmail = await findByTestId("userEmail")

    expect(asFragment()).toMatchSnapshot()
    expect(store.getState()).toMatchInlineSnapshot(`
      Object {
        "authentication": Object {
          "user": null,
        },
      }
    `)
  })
})
