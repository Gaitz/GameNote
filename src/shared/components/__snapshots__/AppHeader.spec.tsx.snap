// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader component Sign out 2`] = `
<DocumentFragment>
  <div
    class="css-yeeskf"
  >
    <div
      class="css-1tv7vc9"
    >
      <h2
        class="chakra-heading css-eedfcw"
      >
        Game Note
      </h2>
    </div>
    <button
      aria-controls="menu-list-2"
      aria-expanded="false"
      aria-haspopup="menu"
      class="chakra-button chakra-menu__menu-button css-1so1lpk"
      data-testid="appHeaderMenuButton"
      id="menu-button-2"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="css-r6z5ec"
      data-popper-escaped=""
      data-popper-placement="bottom-start"
      data-popper-reference-hidden=""
      style="visibility: hidden; min-width: max-content; inset: 0 auto auto 0; margin: 0px; --popper-transform-origin: top left; position: absolute; left: 0px; top: 0px; transform: translate(0px, 8px);"
    >
      <div
        aria-orientation="vertical"
        class="chakra-menu__menu-list css-zenlc5"
        id="menu-list-2"
        role="menu"
        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: visible; transform: scale(0.8) translateZ(0);"
        tabindex="-1"
      >
        <div
          class="chakra-menu__group"
          role="group"
        >
          <p
            class="chakra-menu__group__title css-1a50x1j"
          >
            profile
          </p>
          <div
            class="chakra-stack css-bn6sxo"
          >
            <span
              class="chakra-avatar css-nxs594"
            >
              <svg
                aria-label=" avatar"
                class="chakra-avatar__svg css-16ite8i"
                role="img"
                viewBox="0 0 128 128"
              >
                <path
                  d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                  fill="currentColor"
                />
                <path
                  d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                  fill="currentColor"
                />
              </svg>
            </span>
            <p
              class="chakra-text css-0"
              data-testid="userEmail"
            />
          </div>
        </div>
        <div
          class="chakra-menu__group"
          role="group"
        >
          <p
            class="chakra-menu__group__title css-1a50x1j"
          >
            feature
          </p>
          <button
            class="chakra-menu__menuitem css-1o9rabo"
            data-index="0"
            id="menu-list-2-menuitem-3"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            workoutFeature
          </button>
        </div>
        <div
          class="chakra-menu__group"
          role="group"
        >
          <p
            class="chakra-menu__group__title css-1a50x1j"
          >
            system
          </p>
          <button
            class="chakra-menu__menuitem css-1o9rabo"
            data-index="1"
            data-testid="signOutButton"
            id="menu-list-2-menuitem-4"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            signOut
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AppHeader component Snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-yeeskf"
  >
    <div
      class="css-1tv7vc9"
    >
      <h2
        class="chakra-heading css-eedfcw"
      >
        Game Note
      </h2>
    </div>
    <button
      aria-controls="menu-list-2"
      aria-expanded="false"
      aria-haspopup="menu"
      class="chakra-button chakra-menu__menu-button css-1so1lpk"
      data-testid="appHeaderMenuButton"
      id="menu-button-2"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="css-r6z5ec"
      data-popper-escaped=""
      data-popper-placement="bottom-start"
      data-popper-reference-hidden=""
      style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 8px); --popper-transform-origin: top left;"
    >
      <div
        aria-orientation="vertical"
        class="chakra-menu__menu-list css-zenlc5"
        id="menu-list-2"
        role="menu"
        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
        tabindex="-1"
      >
        <div
          class="chakra-menu__group"
          role="group"
        >
          <p
            class="chakra-menu__group__title css-1a50x1j"
          >
            profile
          </p>
          <div
            class="chakra-stack css-bn6sxo"
          >
            <span
              class="chakra-avatar css-nxs594"
            >
              <svg
                aria-label=" avatar"
                class="chakra-avatar__svg css-16ite8i"
                role="img"
                viewBox="0 0 128 128"
              >
                <path
                  d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                  fill="currentColor"
                />
                <path
                  d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                  fill="currentColor"
                />
              </svg>
            </span>
            <p
              class="chakra-text css-0"
              data-testid="userEmail"
            />
          </div>
        </div>
        <div
          class="chakra-menu__group"
          role="group"
        >
          <p
            class="chakra-menu__group__title css-1a50x1j"
          >
            feature
          </p>
          <button
            class="chakra-menu__menuitem css-1o9rabo"
            data-index="0"
            id="menu-list-2-menuitem-3"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            workoutFeature
          </button>
        </div>
        <div
          class="chakra-menu__group"
          role="group"
        >
          <p
            class="chakra-menu__group__title css-1a50x1j"
          >
            system
          </p>
          <button
            class="chakra-menu__menuitem css-1o9rabo"
            data-index="1"
            data-testid="signOutButton"
            id="menu-list-2-menuitem-4"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            signOut
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
